pkg/buildpipeline/pipeline.go

```diff
@@ -645,44 +669,52 @@ func stageToTask(s Stage, pipelineIdentifier string, buildIdentifier string, nam
 				return nil, errors.New("syntactic sugar steps not yet supported")
 			}
 		}
-
-		return &TransformedStage{Stage: s, Task: t, Depth: depth}, nil
+		ts := TransformedStage{Stage: s, Task: t, Depth: depth, EnclosingStage: enclosingStage, PreviousSiblingStage: previousSiblingStage}
```

Comment:
This and the next line could be combined by switching to `return &TransformedStage{...}`, fwiw.